src/scenic/syntax/compiler.py

```diff
@@ -61,6 +71,75 @@ def visit_Attribute(self, node):
         self.visit(val)
 
 
+class LocalFinder(ast.NodeVisitor):
+    """Utility class for finding all local variables of a code block."""
+
+    @staticmethod
+    def findIn(block):
+        lf = LocalFinder()
+        for statement in block:
+            lf.visit(statement)
+        names = lf.names
+        return names - lf.globals - lf.nonlocals
+
+    def __init__(self):
+        self.names = set()
+        self.globals = set()
+        self.nonlocals = set()
+
+    def visit_Global(self, node):
+        self.globals.update(node.names)
+
+    def visit_Nonlocal(self, node):
+        self.nonlocals.update(node.names)
+
+    def visit_FunctionDef(self, node):
+        self.names.add(node.name)
+        self.visit(node.args)
+        for decorator in node.decorator_list:
+            self.visit(decorator)
+        if node.returns is not None:
+            self.visit(node.returns)
+        # do not visit body; it's another block
+
+    def visit_Lambda(self, node):
+        self.visit(node.args)
+        # do not visit body; it's another block
+
+    def visit_ClassDef(self, node):
+        self.names.add(node.name)
+        for child in itertools.chain(node.bases, node.keywords, node.decorator_list):
+            self.visit(child)
+        # do not visit body; it's another block
+
+    def visit_Import(self, node):
+        for alias in node.names:
+            bound = alias.asname
+            if bound is None:
+                bound = alias.name
+            self.names.add(bound)
+
+    def visit_ImportFrom(self, node):
+        self.visit_Import(node)
+
+    def visit_Name(self, node):
+        if isinstance(node.ctx, (ast.Store, ast.Del)):
+            self.names.add(node.id)
+
+    def visit_ExceptHandler(self, node):
+        if node.name is not None:
+            self.names.add(node.name)
+        self.generic_visit(node)
+
+
+def unquote(s: str) -> str:
```

Comment:
The parser uses `token.string` to create the `docstring` field of `BehaviorDef` and `MonitorDef`. This includes quotation marks and we need to get rid of them at the compiler level (this can probably be done at the parser level, but I don't want to make the grammar too complex by writing a Python helper function).

This `unquote` function looks for four possible patterns for defining strings (`"` `'` `'''` and `"""`) and extract the body.